bpf/lib/conntrack.h

```diff
@@ -254,14 +255,15 @@ static __always_inline __u8 __ct_lookup(const void *map, struct __ctx_buff *ctx,
 			}
 			break;
 		case ACTION_CLOSE:
+		case ACTION_RESET:
 			/* RST or similar, immediately delete ct entry */
 			if (dir == CT_INGRESS)
 				entry->rx_closing = 1;
 			else
 				entry->tx_closing = 1;
 
 			*monitor = TRACE_PAYLOAD_LEN;
-			if (ct_entry_alive(entry))
+			if (ct_entry_alive(entry) && action != ACTION_RESET)
```

Comment:
Doesn't this allow for anybody to spam a ton of RSTs (with invalid sequence numbers) in one direction without requiring the victim to have a chance to react and cause all lifetimes of the conntrack table to be reset to 10s to trigger the expiration of connections that could still be alive?

Reply:
The packets would have to be mapped to the same CT entry, so AFAIU they would need to have the same 4-tuple.

When we originally discussed this approach on the datapath meeting, I believe that @joestringer mentioned that we do not allow pods to send packets that do not match their IP. So the spurious packets will have to come from the same pod as the alive connections, which under normal operation should not happen since the port should already be binded if an active connection exists.

Reply:
If I'm missing something, one way to amortize the problem would be to only set `ACTION_RESET` for CT entries that have not seen both `SYN`s, since from a quick look we are already tracking this. In this case, the issue would happen only during the 3-way handshake.

Reply:
> When we originally discussed this approach on the datapath meeting, I believe that @joestringer mentioned that we do not allow pods to send packets that do not match their IP. So the spurious packets will have to come from the same pod as the alive connections, which under normal operation should not happen since the port should already be binded if an active connection exists.

Agreed but I don't think that is protection enough to assume that we are fully OK with anybody with access to schedule a host networking pod with raw socket access can terminate connections of other pods at will.

> If I'm missing something, one way to amortize the problem would be to only set ACTION_RESET for CT entries that have not seen both SYNs, since from a quick look we are already tracking this. In this case, the issue would happen only during the 3-way handshake.

Yes, that is a good protection mechanism. The handshake makes it reasonable to expect packets delivered in both directions so even if the lifetime is adjusted incorrectly by a malicious packet, it would get reset when the SYN-ACK arrives. The main concern is that the lifetime can be reduced on connections which are iding.

Reply:
Updated version does what is described above, i.e., only reduces the timeout for connections where not both SYNs have been seen.